tests/jballerina-unit-test/src/test/resources/test-src/expressions/conversion/native-conversion-stampable-values.bal

```diff
@@ -104,6 +104,25 @@ function testConvertMapJsonWithDecimalUnionTarget() {
     assert(castedValue["name"], mp["name"]);
 }
 
+public type Scalar int|string|float|boolean;
+
+public type Argument record {|
+    Scalar value;
+|};
+
+public function testConvertUnion() {
+    json expectedJson = { "value": 132 };
+
+    Argument expected = { "value": 132 };
+    var actual = expectedJson.cloneWithType(Argument);
```

Comment:
Were we able to reproduce the original CCE mentioned in https://github.com/ballerina-platform/ballerina-lang/issues/26632? Both this example and the one in the issue seem to result in "ballerina/lang.typedesc}ConversionError" errors on slp7.

Reply:
CCE is not reproducible. Found a different bug and fixed it.

Reply:
I tried the same on slp3 also but still couldn't reproduce the CCE. Fails with the following which I believe has/is been fixed.

```cmd
error: {ballerina/lang.typedesc}ConversionError message='map<json>' value cannot be converted to 'Argument'
        at ballerina.lang_value.1_0_0:cloneWithType(value.bal:67)
           xyz:main(xyz.bal:174)
```

But since the CCE has been observed shall we keep that issue open to try and reproduce the CCE and fix it, and create a new issue for what we're fixing here? Let's also add a comment on the issue to say the CCE is not reproducible with that code.

Reply:
+1. Will do that